modules/swagger-codegen/src/main/resources/typescript-fetch/api.mustache

```diff
@@ -0,0 +1,309 @@
+// tslint:disable
+{{>licenseInfo}}
+
+import * as url                                 	    from "url";
+import * as isomorphicFetch                     	    from "isomorphic-fetch";
+import { Configuration }                                from "./configuration";
+
+const BASE_PATH = "{{{basePath}}}".replace(/\/+$/, "");
+
+/**
+ *
+ * @export
+ * @interface FetchAPI
+ */
+export interface FetchAPI {
+    (url: string, init?: any): Promise<any>;
+}
+
+/**
+ *  
+ * @export
+ * @interface FetchArgs
+ */
+export interface FetchArgs {
+    url: string;
+    options: any;
+}
+
+/**
+ * 
+ * @export
+ * @class BaseAPI
+ */
+export class BaseAPI {
+    public config: Configuration = new Configuration();
```

Comment:
Maybe we can delay the instantiation of the `Configuration` object for after we've verified that the `config` parameter has not been passed.